odoo/orm/domains.py

```diff
@@ -450,7 +455,7 @@ def _optimize(self, model: BaseModel, level: OptimizationLevel) -> Domain:
         while domain._opt_level < level:
             if (count := count + 1) > MAX_OPTIMIZE_ITERATIONS:
                 raise RecursionError("Domain.optimize: too many loops")
-            next_level = OptimizationLevel(domain._opt_level + 1)
+            next_level = domain._opt_level.next_level
```

Comment:
I suggest to use plain integers instead, so that `domain._opt_level + 1` is perfectly legit, and the various constants simply define each level's semantics.  For the typing, you may define `OptimizationLevel` as `typing.Literal[NONE, BASIC, DYNAMIC_VALUES, FULL]`...